DataTable: Add prop for sort direction on first time click

sethu1504 opened the pull request:
Fixes #1768

### Additional description
When the sorting direction is null, the column is sortable and the column is not is not sorted, the first click makes a default sort in ascending.
A prop name 'firstSortDirection' is added to specify the direction the first sort should be if the above-mentioned conditions are satisfied

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

interactivellama commented:
@sethu1504 Following via email.

interactivellama commented:
Fixes #1768

interactivellama commented:
@sethu1504 I updated the boolean prop to `isDefaultSortDescending` to be a little more idiomatic and added an example to a Storybook Story. The logic is the same. Thank you for the contribution!

sethu1504 commented:
Hi Stephen,
Thanks a lot :)

On Wed, Feb 27, 2019 at 3:25 PM Stephen James <notifications@github.com>
wrote:

> @sethu1504 <https://github.com/sethu1504> I updated the boolean prop to
> isDefaultSortDescending to be a little more idiomatic and added an
> example to a Storybook Story. The logic is the same. Thank you for the
> contribution!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/salesforce/design-system-react/pull/1748#issuecomment-468071990>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AblYaKVcxogKWGfyKL33XuakqmoB0KeRks5vRxPvgaJpZM4a6Og7>
> .
>


-- 
Regards,
Sethuraman.
